lambdas/issuecredential/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/pact/utils/PreLambdaHandler.java

```diff
@@ -0,0 +1,245 @@
+
+        Headers serverResponseHeaders = exchange.getResponseHeaders();
+        response.getHeaders().forEach(serverResponseHeaders::set);
+
+        if (!response.getBody().isEmpty()) {
+            LOGGER.info("getting response body");
+
+            String body = response.getBody();
+            try {
+                JWT jwt = JWTParser.parse(body);
+                String jtiId = "urn:uuid:b07cc7e3-a2dc-4b17-9826-6907fcf4059a";
+
+                LocalDate futureDate = LocalDate.of(2099, 1, 1);
+                Date nbf = Date.valueOf(futureDate);
+
+                JWTClaimsSet claimsSet = jwt.getJWTClaimsSet();
+                Object vcClaim = claimsSet.getClaim("vc");
+                Map<String, Object> vc = (Map<String, Object>) vcClaim;
+                Map<String, Object> credentialSubject =
+                        (Map<String, Object>) vc.get("credentialSubject");
+                List<Object> evidence = (List<Object>) vc.get("evidence");
+                List<Object> sortedEvidence =
+                        evidence.stream().sorted().collect(Collectors.toList());
+                TreeMap<Object, Object> sortedCredentialSubject = new TreeMap<>(credentialSubject);
+
+                vc.put("credentialSubject", sortedCredentialSubject);
+                vc.put("evidence", sortedEvidence);
+
+                SignedJWT signedJWT = amendClaimSet(jwt, nbf, jtiId, vc, claimsSet);
+
+                body = signedJWT.serialize();
+            } catch (ParseException e) {
+
+                // Continue
+            }
+
+            exchange.sendResponseHeaders(statusCode, body.length());
+            try (OutputStream os = exchange.getResponseBody()) {
+                os.write(body.getBytes());
+            }
+        }
+    }
+
+    private SignedJWT amendClaimSet(
+            JWT jwt, Date nbf, String jtiId, Map<String, Object> vc, JWTClaimsSet claimsSet)
+            throws JOSEException, ParseException, JsonProcessingException {
+
+        removeNull(vc);
+
+        JWTClaimsSet modifiedClaimsSet =
+                new JWTClaimsSet.Builder(claimsSet)
+                        .notBeforeTime(nbf)
+                        .jwtID(jtiId)
+                        .claim("vc", vc)
+                        .build();
+        SignedJWT signedJWT = new SignedJWT((JWSHeader) jwt.getHeader(), modifiedClaimsSet);
+        signedJWT.sign(signer);
+
+        TreeMap<String, Object> test = new TreeMap<>();
+        test.put("typ", jwt.getHeader().getType().getType());
+        test.put("alg", jwt.getHeader().getAlgorithm().getName());
+
+        Base64URL jwtHeader = Base64URL.encode(new ObjectMapper().writeValueAsString(test));
+        String[] serialize = signedJWT.serialize().split("\\.");
+
+        SignedJWT signedJWTNew =
+                new SignedJWT(
+                        jwtHeader, Base64URL.from(serialize[1]), Base64URL.from(serialize[2]));
+
+        return signedJWTNew;
+    }
+
+    private void removeNull(Map map) {
```

Comment:
Can this be removed and fixed the same way as fraud?

Reply:
Updated @gds-jbentham